google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/RepositorySelector.java

```diff
@@ -0,0 +1,292 @@
+  }
+
+  @Override
+  protected int getPreferredPopupHeight() {
+    return SELECTOR_HEIGHT;
+  }
+
+  private class ProjectNotSelectedPanel extends JPanel {
+
+    private static final int HEIGHT = 30;
+
+    ProjectNotSelectedPanel() {
+      setPreferredSize(new Dimension(RepositorySelector.this.getTextField().getWidth(), HEIGHT));
+      JLabel warning = new JBLabel();
+      warning.setFont(new Font(getFont().getFontName(), Font.ITALIC, getFont().getSize()));
+      warning.setText(GctBundle.message("cloud.repository.selector.missing.project.error"));
+      add(warning);
+    }
+  }
+
+  private class RepositoryPanel extends JPanel {
+
+    private JTree repositoryTree;
+    private DefaultMutableTreeNode projectRootNode;
+    private DefaultTreeModel treeModel;
+    private ProjectRepositoriesModelItem repositories;
+
+    RepositoryPanel(String cloudProject) {
+      repositories
+          = new ProjectRepositoriesModelItem(cloudProject, user);
+      projectRootNode = new DefaultMutableTreeNode("root");
+      treeModel = new DefaultTreeModel(projectRootNode);
+
+      setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
+
+      repositoryTree = new JTree(treeModel);
+      repositoryTree.setRowHeight(0);
+      repositoryTree.setRootVisible(false);
+      repositoryTree.setOpaque(false);
+      repositoryTree.getSelectionModel().setSelectionMode(TreeSelectionModel.SINGLE_TREE_SELECTION);
+      repositoryTree.expandRow(0);
+
+      RepositorySelectorRenderer renderer = new RepositorySelectorRenderer();
+      renderer.setOpenIcon(GoogleCloudToolsIcons.CLOUD);
+      renderer.setClosedIcon(GoogleCloudToolsIcons.CLOUD);
+      renderer.setLeafIcon(null);
+      repositoryTree.setCellRenderer(renderer);
+
+      repositoryTree.addTreeSelectionListener(event -> {
+        DefaultMutableTreeNode node
+            = (DefaultMutableTreeNode) repositoryTree.getLastSelectedPathComponent();
+        if (node != null && node instanceof RepositoryModelItem) {
+          RepositoryModelItem repoNode = (RepositoryModelItem) node;
+          RepositorySelector.this.setText(repoNode.getRepositoryId());
+          ApplicationManager.getApplication().invokeLater(RepositorySelector.this::hidePopup);
+        }
+      });
+
+      JBScrollPane scrollPane = new JBScrollPane();
+      scrollPane.setPreferredSize(new Dimension(getPopupWidth(), getPreferredPopupHeight()));
+      scrollPane.setViewportView(repositoryTree);
+      scrollPane.setBorder(BorderFactory.createEmptyBorder());
+
+      add(scrollPane);
+
+      JPanel bottomPane = new JPanel();
+      JPanel buttonPanel = new JPanel();
+
+      bottomPane.setLayout(new BorderLayout());
+      bottomPane.setBorder(BorderFactory.createMatteBorder(1, 0, 0, 0, Color.LIGHT_GRAY));
+      buttonPanel.setLayout(new BoxLayout(buttonPanel, BoxLayout.LINE_AXIS));
+      buttonPanel.setBorder(BorderFactory.createEmptyBorder(5, 5, 5, 5));
+
+      if (canCreateRepository) {
```

Comment:
then it just omits the "create repository" button. this is as expected - there are 2 dialogs where this widget is used, 1 where we need the button, and 1 where we don't.